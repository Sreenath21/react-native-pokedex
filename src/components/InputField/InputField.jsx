import React, {useState, useEffect} from 'react';
import {View, Text, TextInput, Pressable} from 'react-native';

import styles from './InputField.style';

const InputField = ({setSearchTerm, setModalVisible}) => {
  const [value, setValue] = useState('');

  useEffect(() => {
    setSearchTerm(value.toLowerCase());
  }, [value]);

  const onTextChange = text => {
    setValue(text);
  };

  return (
    <View style={styles.container}>
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.input}
          placeholder="Name or Number"
          keyboardType="default"
          onChangeText={onTextChange}
          value={value}
        />
      </View>
      <Pressable style={styles.filter} onPress={() => setModalVisible(true)}>
        <Text style={{color: '#fff'}}>Filters</Text>
      </Pressable>
    </View>
  );
};

export default InputField;
