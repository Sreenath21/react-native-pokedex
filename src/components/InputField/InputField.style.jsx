import {StyleSheet} from 'react-native';
import COLORS from '../../utils/colors';

const styles = StyleSheet.create({
  container: {
    marginTop: 32,
    marginBottom: 24,
    height: 60,
    flexDirection: 'row',
  },
  inputContainer: {
    backgroundColor: '#dbe4ff',
    flex: 3,
    marginRight: 24,
    borderRadius: 10,
  },
  input: {
    flex: 3,
    fontSize: 16,
    paddingHorizontal: 16,
  },
  filter: {
    flex: 1,
    backgroundColor: COLORS.primary,
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default styles;
