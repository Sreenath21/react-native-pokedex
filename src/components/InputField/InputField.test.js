import React from 'react';
import InputField from './InputField';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

it('InputField renders correctly', () => {
  const tree = renderer.create(
    <InputField setModalVisible={jest.fn()} setSearchTerm={jest.fn()} />,
  );
  expect(tree).toMatchSnapshot();
});

it('InputField should update on search term changes', () => {
  const setSearchTerm = jest.fn();
  const screen = render(
    <InputField setModalVisible={jest.fn()} setSearchTerm={setSearchTerm} />,
  );

  fireEvent.changeText(
    screen.getByPlaceholderText('Name or Number'),
    'pikachu',
  );
  expect(setSearchTerm).toBeCalled();
});
