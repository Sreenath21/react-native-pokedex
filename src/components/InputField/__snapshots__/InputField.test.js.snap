// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "height": 60,
      "marginBottom": 24,
      "marginTop": 32,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#dbe4ff",
        "borderRadius": 10,
        "flex": 3,
        "marginRight": 24,
      }
    }
  >
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Name or Number"
      style={
        {
          "flex": 3,
          "fontSize": 16,
          "paddingHorizontal": 16,
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#391363",
        "borderRadius": 10,
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Filters
    </Text>
  </View>
</View>
`;
